zng/docs/spec.md

```diff
@@ -335,6 +346,46 @@ subsequent key for values that have previous keys of equal value.
 It is an error for any such values to appear that contradicts the most
 recent ordering directives.
 
+### 2.1.3 Frame Markers
+
+A frame marker divides a ZNG stream into a sequence of smaller frames
+which can be decoded indepenently.  It is left up to implementations to
+decide how frequently to use these markers.
+
+A frame marker is encoded as follows:
+```
+----------------
+|0x86|<prevlen>|
+----------------
+```
+where `<prevlen>` is a `uvarint` encoded count of the total number of
+bytes in the preceding frame.  After this marker, all previously read
+typedefs are invalidated and the "next available type ID" is reset to
+the initial value of 23.  To represent subsequent records that use a
+previously defined type, the appropriate typedef control code must
+be re-emitted
+(and note that the typedef may now be assigned a different ID).
+
+The value of `<prevlen>` may be 0 for the first frame in a stream
+or if the length of the previous frame is unknown.  If it is non-zero
+then the contents of the stream at the point `<prevlen>` bytes before
+this marker must also be a frame marker.
+
+Implementations are encouraged to include the previous frame length
+as it allows a BZNG processor to read a large stored BZNG stream backward.
+To do this, a frame marker must be placed at the very end of the
+stream so that a reader may seek to the end of the stream and read the
+final marker and then use `<prevlen>` to find the last frame that holds
+records.  The contents of the each frame must be read in the order they
+appear in the stream.
+
+To process every individual record in a stream in reverse order,
```

Comment:
It might be worth clarifying that the "reverse order" here is unrelated to the orthogonal to the order of "ordering hints" from the preceding section.